src/Test.UnitTests.Sarif.WorkItems/SarifWorkItemExtensionsTests.cs

```diff
@@ -52,26 +53,109 @@ public void SarifWorkItemExtensions_CreateWorkItemTitle_HandlesSingleResultWithR
             sb.Length.Should().Be(0, because: Environment.NewLine + sb.ToString());
         }
 
+        private class PhraseToolNamesTestCase
+        {
+            public PhraseToolNamesTestCase(List<string> input, string expectedOutput)
+            {
+                Input = input;
+                ExpectedOutput = expectedOutput;
+            }
+
+            public List<string> Input { get; }
+            public string ExpectedOutput { get; }
+        }
+
+        private static readonly ReadOnlyCollection<PhraseToolNamesTestCase> s_phraseToolNamesTestCases =
+            new ReadOnlyCollection<PhraseToolNamesTestCase>(new PhraseToolNamesTestCase[]
+            {
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {""},
+                    expectedOutput: ""),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"CredentialScanner"},
+                    expectedOutput: "'CredentialScanner'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"CredentialScanner", "Binskim"},
+                    expectedOutput: "'CredentialScanner' and 'Binskim'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"SEMMLE", "CredentialScanner", "Binskim"},
+                    expectedOutput: "'SEMMLE', 'CredentialScanner' and 'Binskim'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"SEMMLE", "", "Binskim"},
+                    expectedOutput: "'SEMMLE' and 'Binskim'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"", "CredentialScanner", "SEMMLE", "Binskim"},
+                    expectedOutput: "'CredentialScanner', 'SEMMLE' and 'Binskim'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"", "CredentialScanner"},
+                    expectedOutput: "'CredentialScanner'"),
+            });
```

Comment:
>} [](start = 12, length = 1)

Nice set of test cases. I suggest you handle the `null` case as well as the empty string case. SARIF requires `tool.driver.name` to be present, but if it's absent and the file hasn't been through the validator, it can slip through.